### PR TITLE
Remove RM_AutoMemory from search 

### PR DESCRIPTION
**Pull request by ashtul**
and wrap with a macro that checks it is not `RSDummyCtx` on the coordinator.

### CODE REVIEW

File: src/hybrid_reader.c [hunk 1]
```diff
@@ -249,6 +249,7 @@ static void prepareResults(HybridIterator *hr) {
     VecSimQueryResult_Free(hr->list);
     if (hr->iter) {
       VecSimQueryResult_IteratorFree(hr->iter);
+      hr->iter = NULL;
```
Comment (ashtul) [inline src/hybrid_reader.c:252]: Double free if reached `break` on line 257

**Review by rafie — DISMISSED** [review]

File: src/module.h [hunk 2]
```diff
@@ -21,6 +22,12 @@ extern RedisModuleCtx *RSDummyContext;
 /** Indicates that RediSearch_Init was called */
 extern int RS_Initialized;
 
+#define RS_AutoMemory(ctx)                      \
```
Comment (MeirShpilraien) [inline src/module.h:25]: Why you need this then?

Comment (MeirShpilraien) [reply src/module.h:25]: If its only for legacy gc lets drop legacy gc as well

File: src/redisearch.h [hunk 3]
```diff
@@ -13,6 +13,8 @@ typedef uint64_t t_offset;
 // to produce a field mask we calculate 2^fieldId
 typedef uint16_t t_fieldId;
 
+// extern void *RSDummyContext;
```
Comment (MeirShpilraien) [inline src/redisearch.h:16]: remove?

File: src/suggest.c [hunk 4]
```diff
@@ -245,7 +263,7 @@ int parseSuggestOptions(RedisModuleString **argv, int argc, SuggestOptions *opti
 }
 
 int RSSuggestGetCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-  RedisModule_AutoMemory(ctx); /* Use automatic memory management. */
+  RS_AutoMemory(ctx); /* Use automatic memory management. */
```
Comment (MeirShpilraien) [inline src/suggest.c:266]: I thought we want to remove auto memory entirely no?

**Review by MeirShpilraien — COMMENTED** [review]
Don't we want to remove auto memory entirely? If so why we still us it?

File: src/module.h (hunk 2 above)
Comment (ashtul) [reply src/module.h:25]: It is for the coordinator.

**Comment by ashtul** [comment]
> Don't we want to remove auto memory entirely? If so why we still us it?

For the time being, we cannot until we run nightly sanitizer.

**Review by MeirShpilraien — APPROVED** [review]

**Pull request merged**